### ISSUE TITLE
Interactive mode not working in 1.4

### ISSUE THREAD

**Issue by Nodd** [matplotlib/matplotlib#3505]
I have an application with a GUI using Pyside or PyQt (the compatibility  layer is https://github.com/ros-visualization/python_qt_binding). This application displays matplotlib plots in interactive mode to avoid problems with the event loop.
It works well with matplotlib 1.3.1, but when I update to 1.4.0 the plots are not shown anymore. I tried upgrading and downgrading matplotlib only, and it confirms that the problem is linked to the update.

If I try to add `show()` after the plot, the window is shown with the warning `QCoreApplication::exec: The event loop is already running`, as if it was in non-interactive mode.

The return value of `isinteractive()` is 0 if I call plt.ion() and False if I cann plt.ioff().

I tried to write a minimal example of the problem but it works fine in the console. I don't recall anything special in my application that would change the behavior.

I use anaconda under Linux (Manjaro).

**Comment by egiovan** [comment]
I may be little bit late, but I have a similar problem.

I have written a script that first creates a figure with some axes, and then is waiting for some experimental data to be ready, it is calling a soap function connected to some webservices that returned when the experimental data were available.

When they are ready it updates the figure with the new data (well this what it was supposed to do).
Actually I couldn't make it works using the show command. The script is supposed to be running forever (it could be stopped with a ^C if needed). 

I added the pause(..) and it worked but of course the figure was not anymore interactive. The sys.ps1 trick actually didn't work.
 
What is really the meaning of show(block=False) if the figure is not showing itself?

Would it be possible to add to the show command an option to start an event loop just for the figure itself?

I had also written some scripts that were doing some calculation, prepared some plots, and then just before the blocking show command they forked a subprocess, the main process stopped, while the child process issued a blocking show command. The idea was to return the terminal to the user while having the event loop and the figure still active. But that stopped working. Well I removed the forking part, so now the user has to close the figure to have the terminal back (not a big problem actually, but I liked how it was before). Suggestions?

Cheers

**Comment by tacaswell** [comment]
@egiovan Is this still the case in 1.4.3?  We backed out the change that broke this in 1.4.1 so the old behaviour should be restored.

If that is not the case, can you open a new issue with a minimal example of what you are doing?

**Comment by kogelnik** [comment]
I don't see the original problem with 1.4.3.

**Comment by egiovan** [comment]
I haven't tried the script that I'm preparing yet.
I just tried this simple script, and it is not working.
I should say that I'm using anaconda, and I have just updated to the latest version.

import sys
sys.ps1 = 'Ciao'
import time
import numpy as np
import matplotlib
matplotlib.use('qt4agg')
import matplotlib.pyplot as plt

x = np.random.randn(10)
print('ready to plot')
plt.plot(x)
plt.draw()
plt.show(block=False)

print('starting to sleep (or working hard)')
time.sleep(10)
plt.plot(x + 2)
plt.draw()
plt.show(block=False)

print('sleeping again (or more work)')
time.sleep(10)
print('now blocking until the figure is closed')
plt.show(block=True)

The plot is shown only at the end.
The version of matplot lib is 1.4.3
With numpy version 1.6

If there are other test that I could do, tell me.

I may do some other test the next week, but if you have any suggestion they are welcome.
I'll tell you that I'm using SOAPpy to connecting to a blocking webservice. If the pause works it could be ok as the plot should be shown on a screen.

**Comment by WeatherGod** [comment]
If you are using anaconda and have the latest updates, then I doubt you are
using numpy version 1.6, which is rather ancient. Is that a typo and you
meant 1.9?

**Comment by kogelnik** [comment]
I tried the script on 1.4.3 with the Qt backend statement commented out and didn't see any problems.

So possibly it has something to do with the backend.

My numpy is 1.9.2.

**Comment by tacaswell** [comment]
@kogelnik  What kackend are you using?

**Comment by kogelnik** [comment]
TkAgg

I don't have Qt installed.

**Comment by egiovan** [comment]
Yes, sorry, my numpy version was 1.9.2. I fond 1.6 when  I was looking for the matplotlib version:
I typed help(matplotlib) and I searched for a string 'version' in the help.
Basically I found:
matplotlib.__version__ = '1.4.3'
matplotlib.__version__numpy__ = '1.6'

By the way I may have misunderstood the meaning of plt.show(), I tought that it could spawn a thread with an event loop to manage the events in the figure while a main thread can still be active in the current program.

As I told you my problem is  to have a figure with an event loop running and a separated thread that when receives data can update  some figure properties, like some data in some plot.
Which is the easiest and the best way to achieve this?
